test/test_utils/ec2.py

```diff
@@ -509,6 +513,46 @@ def enforce_IMDSv2(instance_id, region=DEFAULT_REGION, ec2_client=None, hop_limi
     LOGGER.info(f"Modify Metadata options State of EC2 instance: {state}")
 
 
+
+def enforce_IMDSv1(instance_id, region=DEFAULT_REGION, ec2_client=None):
+    """
+    Enabled IMDSv1 on EC2 instance.
+
+    :param instance_id: str, ec2 instance id
+    :param region: str, Region where ec2 instance is launched.
+    :param ec2_client: str, ec2 client.
+    :param hop_limit: str, hop limit to be set on ec2 instance.
+    """
+    ec2_client = ec2_client or get_ec2_client(region)
+    response = ec2_client.modify_instance_metadata_options(
+        InstanceId=instance_id,
+        HttpTokens='optional',
+    )
+
+    if not response:
+        raise Exception("Unable to enforce IMDSv1. No response received.")
+
+    timeout = 3
+    state = None
+    if response["InstanceId"]:
+        while timeout > 0:
+            time.sleep(timeout)
```

Comment:
nit: Next time, it would be better to use a `retry` decorator from `tenacity` on a helper function that checks if the metadata is set to the expected state.

While this code also works, and can be merged, it re-implements a solution to an already solved problem.

Reply:
yes, made change